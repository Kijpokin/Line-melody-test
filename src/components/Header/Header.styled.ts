import styled from '@emotion/styled'

export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  min-height: 100px;
  background: #00b900;
  color: #ffffff;
`
