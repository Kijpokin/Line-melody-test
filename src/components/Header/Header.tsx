import { Container } from '@components/styled/Container.styled'
import * as React from 'react'
import { Wrapper } from './Header.styled'

type HeaderProps = {
  children: React.ReactNode
}

const Header: React.FC<HeaderProps> = ({ children }) => (
  <Wrapper>
    <Container>{children}</Container>
  </Wrapper>
)

export default Header
