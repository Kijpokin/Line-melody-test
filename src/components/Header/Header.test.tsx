import { render } from '@testing-library/react'
import Header from './Header'

describe('<Header />', () => {
  it('should match snapshots', () => {
    const mockProps = {
      children: 'Mock text',
    }
    const { container } = render(<Header {...mockProps} />)

    expect(container).toMatchSnapshot()
  })

  it('should render text when provided children', () => {
    const mockProps = {
      children: 'Mock text',
    }
    const { getByText } = render(<Header {...mockProps} />)
    expect(getByText('Mock text')).toBeInTheDocument()
  })
})
