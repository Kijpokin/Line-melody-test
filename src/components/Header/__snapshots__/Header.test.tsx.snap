// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshots 1`] = `
<div>
  <div
    class="css-1br5xam"
  >
    Mock text
  </div>
</div>
`;
